### ISSUE TITLE
About: Bleed-Through Removal from Degraded Documents

### ISSUE THREAD

**Issue by kerberosargos** [ZZZHANG-jx/DocRes#14]
Firstly thank you for your great work. I am newbie about AI. I need your help. 

My question is about "[Bleed-Through Removal from Degraded Documents](https://www.researchgate.net/publication/228516870_Bleed-through_removal_in_degraded_documents)" I have some images for testing. Your Docres "appearance" task almost works perfectly. I would like to clean images than try to detect text with "[Craft](https://github.com/clovaai/CRAFT-pytorch)" model. But some transparent object in image, causes problems. Some samples are as following. Can you give a advice for removing transparent objects for good text detection? 

Thank you in advance for your effort.

<table>
<tr>
<th width="33%">Orginal</th>
<th width="34%">DocRes</th>
<th width="33%">Craft</th>
</tr>
<tr>
<td width="33%"><img src="https://github.com/user-attachments/assets/ee8fdd2a-3772-444b-978a-908db0ff409b"></td>
<td width="34%"><img src="https://github.com/user-attachments/assets/7c358c15-1e8f-4911-9c1c-f56dfed1161e"></td>
<td width="33%"><img src="https://github.com/user-attachments/assets/8feea83e-3541-4091-a34c-aa4aad071815"></td>
</tr>
<table>

<table>
<tr>
<th width="33%">Orginal</th>
<th width="34%">DocRes</th>
<th width="33%">Craft</th>
</tr>
<tr>
<td width="33%"><img src="https://github.com/user-attachments/assets/d4f64c5c-a9bd-44b5-bf6b-d110b195134e"></td>
<td width="34%"><img src="https://github.com/user-attachments/assets/19b441c4-ab28-47dd-aa3a-00ebcae6d9a0"></td>
<td width="33%"><img src="https://github.com/user-attachments/assets/6706a34a-a54a-4da8-9b6c-688347752150"></td>
</tr>
<table>

<tr>
<tr>
<th width="33%">Orginal</th>
<th width="34%">DocRes</th>
<th width="33%">Craft</th>
</tr>
<td width="33%"><img src="https://github.com/user-attachments/assets/571871aa-0983-4bf6-8bc0-3e1222e2cc23"></td>
<td width="34%"><img src="https://github.com/user-attachments/assets/291af31f-fb4e-47a6-b4d3-72403137cd0c"></td>
<td width="33%"><img src="https://github.com/user-attachments/assets/e8b5b5e6-14d7-4caf-a1a0-8206db121e81"></td>
</tr>
<table>

**Comment by ZZZHANG-jx** [comment]
> Firstly thank you for your great work. I am newbie about AI. I need your help.
> 
> My question is about "[Bleed-Through Removal from Degraded Documents](https://www.researchgate.net/publication/228516870_Bleed-through_removal_in_degraded_documents)" I have some images for testing. Your Docres "appearance" task almost works perfectly. I would like to clean images than try to detect text with "[Craft](https://github.com/clovaai/CRAFT-pytorch)" model. But some transparent object in image, causes problems. Some samples are as following. Can you give a advice for removing transparent objects for good text detection?
> 
> Thank you in advance for your effort.

Thank you for your interest in our work. For the bleed-through degradation, we primarily simulated it in our training data using the [bleed_through](https://github.com/ZZZHANG-jx/DocRes/blob/d9f97548faffaacc898aa2bada1ef6bea58ea506/loaders/docres_loader.py#L282) function. In the images you provided, the bleed-through is quite severe. I believe the simulation in the training data needs to be more intense, such as reducing the value in [this line](https://github.com/ZZZHANG-jx/DocRes/blob/d9f97548faffaacc898aa2bada1ef6bea58ea506/loaders/docres_loader.py#L288) from 0.75 to a smaller number, and then fine-tuning the model. Additionally, you might want to try our other work, [GCDRNet](https://github.com/ZZZHANG-jx/GCDRNet), which is specifically designed for appearance enhancement and might yield better results.

**Comment by kerberosargos** [comment]
Firstly thank you very much for your quick answer. Acctually I have tried GCDRNet before DocRes with same test images. But DocRes result is better than GCRNet for me.

As I understand from your answer, I should train again your used dataset by changing   0.75 value. am I right?

**Comment by ZZZHANG-jx** [comment]
> Firstly thank you very much for your quick answer. Acctually I have tried GCDRNet before DocRes with same test images. But DocRes result is better than GCRNet for me.
> 
> As I understand from your answer, I should train again your used dataset by changing 0.75 value. am I right?

Yes, that's correct. If you're not concerned with other tasks (such as binarization or correction), you can train only the appearance task, which will simplify things.

**Comment by foocker** [comment]
> > Firstly thank you very much for your quick answer. Acctually I have tried GCDRNet before DocRes with same test images. But DocRes result is better than GCRNet for me.
> > As I understand from your answer, I should train again your used dataset by changing 0.75 value. am I right?
> 
> Yes, that's correct. If you're not concerned with other tasks (such as binarization or dewarping), you can train only the appearance task, which will simplify things.

hi, how to make the appearance data? your readme.md is not clear. data/preprocess/shadow_extract.py->data/MBD/infer.py->loaders/docres_loader.py
